stores/Todo.store.ts

```diff
@@ -1 +1,52 @@
-class TodoStore {}
+import AsyncStorage from "@react-native-async-storage/async-storage";
+import { action, makeAutoObservable } from "mobx";
+import { Alert } from "react-native";
+
+export type Todo = {
+  title: string;
+  text: string;
+};
+
+export class TodoStore {
+  private _todos: Todo[] = [];
+
+  constructor(todos: Todo[]) {
+    this._todos = todos;
+    makeAutoObservable(this, {
+      setTodos: action,
+    });
+  }
+
+  get todos() {
+    return this._todos;
+  }
+
+  setTodos = (todos: Todo[]) => {
+    this._todos = todos;
+    saveToStorage(todos);
+  };
+
+  addTodo = (todo: Todo) => {
+    this.setTodos([todo, ...this.todos]);
+  };
+}
+
+export async function loadFromStorage() {
```

Comment:
these two functions can be exported to another file (or class) that handle loading and saving. The Todo's gateway.

If you type them correctly (accept Todoo[]) return Todo[], etc. The Store will not have to be concerned with where it saves. It just leaves it to the gateway.

Reply:
Another comment to help you out with structuring your code better.
You have hardcoded these two async functions in your Store. So this is a an AsyncStorageStore.

What if I wanted you to have an BackendAPIStore, which makes a call to API endpoint?
You'd have to go and replace all the calls in the code.

Solution 1) You could define these two functions in a 'gateaway' class and create an instance as private prop. So all other member functions of Store this private prop to save.

Solution 2) Making the solution better you could inject this gateway class in the constructor. So if you wanted to write a unit test for this class, you could 'inject' a test version of the 'gateway' class that does not access async storage (there is none in test environment).

And then you would have to write the logic for injecting either a normal AsyncStorage or a 'FakeAsyncStorage' gateway based on the environment you are. So the use of an IoC container (like inversify becomes interesting ;-) 

You don't have to go all the way to Solution 2). I just wanted to mention here logic behind what we do.

Reply:
Okay, I've moved it to an interface based method, which should be the most flexible without bringing in a big library to handle it for us.